identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/pickLocation/PickLocationScreen.kt

```diff
@@ -0,0 +1,115 @@
+import net.thechance.mena.identity.presentation.components.ErrorSnackBar
+import net.thechance.mena.identity.presentation.screen.enableLocationScreen.EnableLocationScreen
+import net.thechance.mena.identity.presentation.screen.pickLocation.components.EditMapButton
+import net.thechance.mena.identity.presentation.screen.pickLocation.components.GpsFabButton
+import net.thechance.mena.identity.presentation.screen.pickLocation.components.Map
+import org.jetbrains.compose.resources.stringResource
+
+class PickLocationScreen() : BaseScreen<PickLocationScreenViewModel,
+        PickLocationScreenUIState,
+        PickLocationScreenUIEffect,
+        PickLocationScreenInteractionListener>() {
+
+    @Composable
+    override fun Content() {
+        InitScreen(getScreenModel())
+    }
+
+    @Composable
+    override fun OnRender(
+        state: PickLocationScreenUIState,
+        listener: PickLocationScreenInteractionListener
+    ) {
+        Scaffold(
+            topBar = {
+                AuthAppBar(
+                    title = stringResource(Res.string.pick_location_title),
+                    onClickBack = listener::onClickBack,
+                    backgroundColor = Theme.colorScheme.background.surfaceLow
+                )
+            }
+        ) {
+            Map(
+                cameraPosition = state.cameraPosition,
+                onCameraMoved = listener::onMoveCamera,
+                onMapClick = listener::onClickMap,
+                anchorLocation = state.pointerLocation,
+                isLocked = state.isMapLocked,
+                currentLocation = state.currentLocation,
+                animateToCurrentLocation = state.animateToCurrentLocation
+            ) {
+                Column(
+                    Modifier
+                        .padding(16.dp)
+                        .fillMaxSize(),
+                    verticalArrangement = Arrangement.Bottom
+                ) {
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        GpsFabButton(
+                            onClick = listener::onClickGps,
+                            isLoading = state.isGpsButtonLoading,
+                            modifier = Modifier.padding(bottom = 12.dp)
+                        )
+                        EditMapButton(
+                            anchorLocation = state.pointerLocation,
+                            onEditClick = listener::onClickEdit
+                        )
+                    }
+
+                    PrimaryButton(
+                        text = stringResource(Res.string.confirm),
+                        onClick = listener::onClickConfirm,
+                        isEnabled = state.isConfirmEnabled,
+                        isLoading = state.isLoading,
+                        contentPadding = PaddingValues(vertical = 13.dp),
```

Comment:
Even if no 13 in `Theme.spacing`, so 12 would be great
```suggestion
                        contentPadding = PaddingValues(vertical = Theme.spacing._12),
```